text/0077-issue-platform.md

```diff
@@ -0,0 +1,85 @@
+
+# Motivation
+
+The primary purpose of the issue platform is to allow different teams at Sentry to 
+detect problems relevant to our users in their data, and create new issues types from 
+that data.
+
+# High Level Concepts
+
+ - Event: A general term for data sent from SDKs to Sentry. This includes errors, transactions, profiles, replays, etc.
+ - Issue: An aggregation of events representing a unique problem we detected in the user’s system.
+ - Detector: A logical unit that analyzes data looking for problematic patterns and outputs found problems.
+ - Issue Occurrence: An event specifically associated with an issue. These are raised from detectors and sent to Sentry via the occurrences api.
+ - Fingerprint: An identifier of a unique occurrence. It’s used for grouping occurrences into issues.
+ - Issue Category: The broad category that an issue belongs to. `Error`, `Performance` and `Profile` are examples of this.
+ - Issue Type: The specific type of an issue. `N + 1 DB query` is an example of this. Type implies category.
+
+
+# Ingestion Schema
+
+Ingestion Schema
+Ingestion occurs via Kafka. The message format looks like
+
+ - id: str - A uuid representing the issue occurrence id.
+ - project_id: int.
+ - fingerprint: Sequence[str] - The fingerprint to uniquely identify a specific issue
+ - issue_title: str - The title of the occurrence. Will be used as issue title.
+ - subtitle: str - Used as the sub header on the issue.
+ - resource_id: str | None - An optional id that we can use to link to an external system
+ - evidence_data: Mapping[str, Any] - An arbitrary json blob of data for use with the frontend to render any custom components necessary for this issue type.
+ - evidence_display: A list of data to display in the ui and notifications. This will typically be displayed in a tabular format. Each entry is a dict in format
+   - name: str: Name of the value to display in the table
+   - value: str: Value to display in the table
+   - important: bool - Up to one piece of data in evidence_display should be marked as important. This entry will be used in notifications where space is constrained.
+ - type: int - The issue type that this occurrence belongs to. Must be registered as a dataclass like so: https://github.com/getsentry/sentry/blob/5fd8242aca575c84314f1b6744be2ec1d2e9e471/src/sentry/issues/grouptype.py#L143-L148
+ - detection_time: float - Timestamp representing when the occurrence was detected
+ - level: str - The error level we want to display for the occurrence.
+
+One (but not both) of these should also be passed
+ - event_id: str - A uuid representing the event id. Pass this if you want to use an event that has already been stored into nodestore.
+ - event: dict - A dictionary that matches our event format (https://develop.sentry.dev/sdk/event-payloads/). Pass as much of this as makes sense for your issue type. Use this method if your event isn’t already in nodestore.
```

Comment:
Do we actually store this as a dictionary in clickhouse? This would be very inefficient to query.